### DISCUSSION TITLE
Problems using additional parser

### DISCUSSION THREAD

**Discussion by mhanberg** [nvim-treesitter/nvim-treesitter#3367] [Q&A] [answered]
Hi!

I wrote a tree sitter parser https://github.com/mhanberg/tree-sitter-monkey and am attempting to use it with neovim.

I have confirmed that the monkey.so file was created and in the right directory, it seems to think it's installed, but when i load up a monkey file, I get this error.

<img width="1072" alt="image" src="https://user-images.githubusercontent.com/5523984/186774223-631282bc-6274-47d2-9438-a1fcff21c9ca.png">

This is my treesitter config

```lua
local parser_config = require("nvim-treesitter.parsers").get_parser_configs()
parser_config.monkey = {
  install_info = {
    url = "https://github.com/mhanberg/tree-sitter-monkey", -- local path or git repo
    files = { "src/parser.c" },
    -- optional entries:
    branch = "main", -- default branch in case of git repo if different from master
    filetypes = {"monkey"}
  },
}

require("nvim-treesitter.configs").setup {
  ensure_installed = "all",
  ignore_install = { "haskell", "phpdoc" },
  highlight = { enable = true },
  indent = { enable = true },
  playground = {
    enable = true,
    disable = {},
    updatetime = 25, -- Debounced time for highlighting nodes in the playground from source code
    persist_queries = false, -- Whether the query persists across vim sessions
    keybindings = {
      toggle_query_editor = "o",
      toggle_hl_groups = "i",
      toggle_injected_languages = "t",
      toggle_anonymous_nodes = "a",
      toggle_language_display = "I",
      focus_language = "f",
      unfocus_language = "F",
      update = "R",
      goto_node = "<cr>",
      show_help = "?",
    },
  },
}

vim.cmd([[set foldmethod=expr]])
vim.cmd([[set foldexpr=nvim_treesitter#foldexpr()]])
```

Any ideas?

**Comment by theHamsta** [comment, marked as answer]
You need to use `parser_config.Monkey`: https://github.com/mhanberg/tree-sitter-monkey/blob/main/grammar.js#L2

**Reply by mhanberg** [reply]
I figured that might be a problem, but i did actually update that before I saw your reply and it's still not working 😅.

let me try uninstalling and then re-installing instead of updating and seeing what happens.

**Reply by mhanberg** [reply]
Oh, I didn't re-generate. One second

**Reply by mhanberg** [reply]
It's working! Thank you!

**Reply by theHamsta** [reply]
The reason for this is that `nvim` and other tree-sitter language will look for `tree_sitter_XXX` in the grammar you compiled via `(uv_)dlopen`. XXX is determined directly by the same in `grammar.js`. It works like this https://github.com/tree-sitter/tree-sitter-c/blob/3ced8d6cd212a6f576cd4ef3d533bcb9c09eface/src/parser.c#L75605-L75635
